prime-dotnet-webapi/Services/PlrProviderService.cs

```diff
@@ -72,10 +72,12 @@ public async Task<IEnumerable<PlrViewModel>> GetPlrDataByCollegeIdsAsync(IEnumer
                 .ProjectTo<PlrViewModel>(_mapper.ConfigurationProvider, new { plrRoleTypes, plrStatusReasons })
                 .ToListAsync();
 
+            // If a PlrViewModel has ExpertiseCodes, translate the codes to human-readable text
             // PlrProvider's Expertise array does not play well with automapper ProjectTo, map manually before return
             return plr.Select(p =>
                 {
-                    p.Expertise = string.Join(", ", _context.Set<PlrExpertise>().Where(e => p.ExpertiseCode.Contains(e.Code)).Select(e => e.Name));
+                    p.Expertise = string.Join(", ", _context.Set<PlrExpertise>().Where(e =>
+                        (p.ExpertiseCode != null && p.ExpertiseCode.Contains(e.Code))).Select(e => e.Name));
```

Comment:
As a developer unfamiliar with the implementation of PLR, I would expect that `p.ExpertiseCode` cannot be null after a projection as that is the standard behavior when retrieving collections from an ORM (empty collections, never null collections).  

Knowing how this is implemented, I know that the Property has a custom implementation that stores a string collection in a single column. I feel that changing either how that column is materialized so it is never null or at the very least setting the Property to an empty collection when mapped is less surprising solution for the developers.

Reply:
It seems that `NULL` in the `Expertise` database column is propagated to `null` in `PlrProvider.Expertise`.  Then that sets `null` in `PlrViewModel.ExpertiseCode` during the AutoMapper mapping.  Although I added `public ICollection<string> Expertise { get; set; } = new List<string>();` and `public IEnumerable<string> ExpertiseCode { get; set; } = new List<string>();` to initialize the properties in-line, I still get:

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.EntityFrameworkCore.Query.SqlNullabilityProcessor.<VisitIn>g__ProcessInExpressionValues|25_0(SqlExpression valuesExpression, Boolean extractNullValues)
   at Microsoft.EntityFrameworkCore.Query.SqlNullabilityProcessor.VisitIn(InExpression inExpression, Boolean allowOptimizedExpansion, Boolean& nullable)
   at Microsoft.EntityFrameworkCore.Query.SqlNullabilityProcessor.Visit(SqlExpression sqlExpression, Boolean allowOptimizedExpansion, Boolean preserveNonNullableColumns, Boolean& nullable)
   at Microsoft.EntityFrameworkCore.Query.SqlNullabilityProcessor.Visit(SqlExpression sqlExpression, Boolean allowOptimizedExpansion, Boolean& nullable)
   at Microsoft.EntityFrameworkCore.Query.SqlNullabilityProcessor.Visit(SelectExpression selectExpression)
   at Microsoft.EntityFrameworkCore.Query.SqlNullabilityProcessor.Process(SelectExpression selectExpression, IReadOnlyDictionary`2 parameterValues, Boolean& canCache)
   at Npgsql.EntityFrameworkCore.PostgreSQL.Query.Internal.NpgsqlParameterBasedSqlProcessor.ProcessSqlNullability(SelectExpression selectExpression, IReadOnlyDictionary`2 parametersValues, Boolean& canCache)
   at Microsoft.EntityFrameworkCore.Query.RelationalParameterBasedSqlProcessor.Optimize(SelectExpression selectExpression, IReadOnlyDictionary`2 parametersValues, Boolean& canCache)
   at Microsoft.EntityFrameworkCore.Query.Internal.RelationalCommandCache.GetRelationalCommand(IReadOnlyDictionary`2 parameters)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.Enumerator.InitializeReader(DbContext _, Boolean result)
   at Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.NpgsqlExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.Enumerator.MoveNext()
   at System.String.Join(String separator, IEnumerable`1 values)
   at Prime.Services.PlrProviderService.<>c__DisplayClass3_0.<GetPlrDataByCollegeIdsAsync>b__0(PlrViewModel p) in C:\Users\177092\Source\Repos\moh-prime\prime-dotnet-webapi\Services\PlrProviderService.cs:line 79
```

With regards to changing the code in `PlrProviderConfiguration` that populates `PlrProvider.Expertise` from the database, see https://github.com/dotnet/efcore/issues/13850; not sure if removing the null check in `PlrProviderService` for a better implementation is feasible.